pkg/kgo/partitioner.go

```diff
@@ -487,7 +487,21 @@ func KafkaHasher(hashFn func([]byte) uint32) PartitionerHasher {
 	}
 }
 
-// SaramaHasher returns a PartitionerHasher using hashFn that mirrors how
+// SaramaHasher is not compatible with Sarama's default default partitioner.
+// If you need sarama compatibility use SaramaCompatHasher instead.
+// This function is left as is to provide compatibility with older versions of
+// this library.
```

Comment:
Can you change this doc line to:
```
// Deprecated: SaramaHasher is not compatible with Sarama's default partitioner
// and only remains to avoid re-keying records for existing users of this API. See
// [SaramaCompatHasher] for a correct partitioner.
```
?

Then I think this PR is good to go in the next minor release.